src/Storages/RabbitMQ/StorageRabbitMQ.cpp

```diff
@@ -679,13 +677,26 @@ ConsumerBufferPtr StorageRabbitMQ::popReadBuffer(std::chrono::milliseconds timeo
     auto buffer = buffers.back();
     buffers.pop_back();
 
+    if (buffer->needChannelUpdate())
+    {
+        if (buffer->queuesCount() != queues.size())
+            buffer->updateQueues(queues);
+
+        buffer->updateAckTracker();
+
+        if (updateChannel(buffer->getChannel()))
+            buffer->setupChannel();
+    }
+
     return buffer;
 }
 
 
 ConsumerBufferPtr StorageRabbitMQ::createReadBuffer()
 {
-    ChannelPtr consumer_channel = std::make_shared<AMQP::TcpChannel>(connection.get());
+    ChannelPtr consumer_channel;
+    if (event_handler->connectionRunning())
+        consumer_channel = std::make_shared<AMQP::TcpChannel>(connection.get());
```

Comment:
Maybe we should throw an exception otherwise? Seems like we will send `nullptr` into `ReadBufferFromRabbitMQConsumer`?

Reply:
1. Exception is not good here - startup will fail. 
2. nullptr is ok, because then in readPrefix() when poping buffer it will there also check whether channel needs update and will update it if possible. If connection not ok, it will quickly get out of reading phase and streams will be finished.  And reconnecting attempts will be started.